### ISSUE TITLE
Can we chat with 2 departments at the same time?

### ISSUE THREAD

**Issue by vodkhang** [RocketChat/Rocket.Chat.Livechat#209]
**Is your feature request related to a problem? Please describe.**
In our company, we need the customer to be able to communicate with 2 departments at the same time. We already had a specific routing to 1 department like http://localhost:3000/livechat/uTHqz7pxYktvNFwAq

However, if a user accesses to 2 departments at the same time on the same browser, they cannot do that. The data are shared between 2 browser tabs. I think at this point, my fork may start working differently than the original livechat

I think the problem is with a shared localStorage and a shared in memory Mongo Collection

**Describe the solution you'd like**
I am thinking to fix the issue by having a specific local storage for each department id. I think that seems doable. 

I just want to know what is the best way I should do it. And if this will conflict with any future development of our rocketchat's livechat?

**Describe alternatives you've considered**

**Additional context**

**Comment by voordev** [comment]
That would be nice.
I also like the intercom UI for the header very much.
Any plans to restyle the current one like that?

**Comment by hartois** [comment]
Yes, this is a very serious problem. 
We need a client from two different sites to communicate with different departments and these were separate conversations.
Are there any solutions expected?

**Comment by renatobecker** [comment]
> Yes, this is a very serious problem.
> We need a client from two different sites to communicate with different departments and these were separate conversations.
> Are there any solutions expected?

Yeah, there is, you can use the `livechat widget api` and set up the department routing for each website, see:

https://docs.rocket.chat/api/livechat-api#assign-chats-to-a-specific-department

**Comment by hartois** [comment]
Perhaps I do not understand something?
I have two sites: domain1.com and domain2.com. There are two departments: department1 and department2.
I connect a script with on domain1.com
```
this.setDepartment('department1');
```
On the other with
```
this.setDepartment('department2');
```
Next I start a chat from one site and accept it as an agent.
I open a chat on another site and it has already started and the department has been selected. On the second site
```
this.setDepartment(department);
```
does not work and this is the same chat room as on first site. I need these to be different rooms.

**Comment by hartois** [comment]
As I understand it, because the visitor's identification data stored in the cookie.
Need another method or workaround.
